plugins/strategy/action.go

```diff
@@ -70,7 +70,7 @@ func (a *Action) PushReason(r string) {
 		}
 	}
 
-	// Append current reason to history and update action
-	a.Meta[metaKeyReasonHistory] = append(history, a.Reason)
+	// Append current reason to history and update action.
 	a.Reason = r
+	a.Meta[metaKeyReasonHistory] = append(history, a.Reason)
```

Comment:
```suggestion
	if (a.Reason) {
		history = append(history, a.Reason)
	}
	a.Reason = r
	a.Meta[metaKeyReasonHistory] = history
```

Reply:
my only question would be whether we should bother duplicating a.Reason in the reason history?

Reply:
Good shout. I don't have any objection to removing the duplicate. @lgfa29 thoughts?

Reply:
So to get the full reason history you would concatenate `a.Reason + a.Meta[metaKeyReasonHistory]`?

I am ok with that.

Reply:
OK. I will follow up with an additional PR as this changes behaviour rather than fixing bugs/lint uncovered by adding tests.